### ISSUE TITLE
Implementierung des Garage Endpoints mit CRUD-Funktionalität

### ISSUE THREAD

**Issue by Eyad-Wahdan** [remsfal/remsfal-backend#265]
### **Beschreibung:**

Implementiere einen neuen Garage Endpoint, der sowohl die Definitionen im **Core-Package** als auch die konkrete Implementierung im **Service-Package** umfasst. Ziel ist es, alle notwendigen Klassen, Interfaces und Logik bereitzustellen, um vollständige CRUD-Operationen für Garage-Entitäten zu ermöglichen. Die Arbeitsschritte umfassen folgende Aufgaben:

### **Aufgaben im Core-Package:**

1. **Definitionen erstellen und abstimmen**:
    - Implementierung des **GarageEndpoint Interfaces**
    - Implementierung der **GarageJson Klasse**
    - Implementierung des **GarageModel Interfaces**
    - Sicherstellen, dass alle Definitionen die gleichen Attribute beinhalten:
        - `ID (String)`
        - `BUILDING_ID (String)`
        - `TITLE (String)`
        - `LOCATION (AddressEntity)`
        - `DESCRIPTION (String)`
        - `USABLE_SPACE (Float)`

### **Aufgaben im Service-Package:**

1. **CRUD-Funktionalität implementieren**:
    - Implementierung der **GarageResource Klasse** mit REST-Endpunkten für CRUD-Operationen (Create, Read, Update, Delete)
    - Implementierung des **GarageControllers** und der Logik für CRUD-Operationen
    - Implementierung der **GarageRepository Klasse** für Datenbankzugriffe
    - Implementierung der **GarageEntity Klasse** mit den oben genannten Attributen

### **Tests und Validierung:**

- Schreiben von Tests für alle CRUD-Endpunkte in der **GarageResource Klasse**
- Validierung der Funktionalität des **GarageControllers** und der Datenbankoperationen in der **GarageRepository Klasse**

**Comment by F4c3hugg3r** [comment]
Notice that there are already some methods defined in BuildingController and BuildingControllerTest regarding your functionality, which need to be immigrated into your Controller & Test file.

**Comment by Eyad-Wahdan** [comment]
@F4c3hugg3r Thank you for pointing this out! I was actually already aware of it, but I appreciate you taking the time to highlight it 🤝

**Comment by mesyamaureen** [comment]
after discussion, we decided to keep the data type of `LOCATION` as string @astanik . 
Frontend-wise, attributes look good to me.

**Comment by Eyad-Wahdan** [comment]
For your information, the final structure of the `GarageEntity` consists of the following attributes:

- **`ID (String)`**: Unique identifier for the garage
- **`BUILDING_ID (String)`**: ID of the building the garage belongs to
- **`TITLE (String)`**: Name or title of the garage
- **`LOCATION (String)`**: Physical location or address of the garage
- **`DESCRIPTION (String)`**: Optional description of the garage
- **`USABLE_SPACE (Float)`**: Usable space within the garage

Additionally, the `GarageEntity` class inherits from the `RentalUnitEntity` class, which provides the following properties:

- **`PROJECT_ID (String)`**: ID of the project the garage is associated with
- **`TENANCY (TenancyEntity)`**: Optional tenancy details linked to the garage

**Comment by astanik** [comment]
@Eyad-Wahdan An issue is only finished, when the code is completely in the main branch.

**Comment by Eyad-Wahdan** [comment]
@astanik Fair point! Should I first merge my current PR-Branch into main and then create a new Pull Request?
